packages/grafana-ui/src/components/Drawer/Drawer.tsx

```diff
@@ -107,7 +107,17 @@ export function Drawer({
       open={true}
       onClose={onClose}
       placement="right"
-      getContainer={'.main-view'}
+      getContainer={() => {
+        // Use the panes container instead of main, so drawer positions relative to sidebar layout
+        // The panes container has position: relative when sidebar is open, so drawer will position correctly
+        const panesContainer = document.querySelector('.page-panes');
+        const container = panesContainer || document.querySelector('main') || document.body;
+        // Ensure container has position: relative for absolute positioning to work
+        if (container instanceof HTMLElement && getComputedStyle(container).position === 'static') {
+          container.style.position = 'relative';
+        }
```

Comment:
why is this logic needed, from what I can tell  AppChrome should always have the required div 

I cannot find AppChrome rendering a div with `page-panes` class . so for me `document.querySelector('.page-panes');` returns null

Reply:
Yea, correct. Explained why I needed to move to `main` instead of `.main-view` though: https://github.com/grafana/grafana/pull/114304#discussion_r2555483071